### PR TITLE
Update roadmap for v0.5.0

### PR DESCRIPTION
**Pull request by arkodg**
* Moves #24 into v0.5.0 since it carries over from v0.4.0
* Adds #1365 since it tracks the work items of the Scale theme
* Removed other items not tied directly to the roadmap theme
* Added a placeholder roadmap theme for v0.6.0

### CODE REVIEW

**Comment by arkodg** [comment]
ptal @envoyproxy/gateway-maintainers , hoping to get a majority approval on the roadmap

* @zirain @Xunzhuo  #701 is assigned to you, can you please confirm that you plan on working on this issue
* @Xunzhuo #1365 is assigned to you as well , requesting to prioritize #701 over this one :)
* #24 is assigned to me, I plan on spending cycles on this issue

**Comment by zirain** [comment]
Data plane Observability will be fine in v0.5 to me, no bandwidth for control plane.

BTW, I think we should start basic e2e tests in v0.5, WDYT?

**Comment by arkodg** [comment]
thanks for the ACK @zirain . Yes would be good to also start e2e (currently blocked on upstream release) but imo would prioritize data plane observability over it

**Review by qicz — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]
LGTM, I can work on control plane side observation

**Comment by Xunzhuo** [comment]
What e2e test @zirain mentioned for? Do I miss some contexts on it?

**Comment by arkodg** [comment]
> What e2e test @zirain mentioned for? Do I miss some contexts on it?

hey @Xunzhuo its [this](https://github.com/envoyproxy/gateway/issues/863) one, this is a good to have for v0.5.0 but a must have for v0.6.0

**Comment by Xunzhuo** [comment]
Got it thanks ! @arkodg

**Review by zirain — APPROVED** [review]

**Pull request merged**
